File: examples/backend/java-spring/UserController.java
package example;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.Statement;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class UserController {
  private final Connection connection;

  public UserController(Connection connection) {
    this.connection = connection;
  }

  @GetMapping("/users")
  public String users(@RequestParam String role) throws Exception {
    Statement statement = connection.createStatement();
    // Intentional fixture: request parameter is concatenated into SQL.
    ResultSet rs = statement.executeQuery("select email from users where role = '" + role + "'");
    return rs.next() ? rs.getString("email") : "";
  }
}

File: examples/backend/php-laravel/UserController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use Illuminate\Support\Facades\DB;

class UserController extends Controller
{
    public function show(Request $request)
    {
        $id = $request->query('id');
        // Intentional fixture: request parameter is concatenated into SQL.
        return DB::select("select * from users where id = " . $id);
    }
}
